### PR TITLE
[modules][system-ui] Persist user selected background color

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Closes ENG-8736

# How
Setting the background color is now persisted and restored when the app is launched (on iOS). On Android, a new function had to be added to restore the color on app start. If no color is provided the default will be black when in dark mode or white otherwise.

# Test Plan
Tested on physical devices and emulators with bare expo.

### CODE REVIEW

File: packages/expo-system-ui/src/SystemUI.ts
Comment (EvanBacon) [inline packages/expo-system-ui/src/SystemUI.ts:42]: I think you may need to update the versioned docs(?) not sure how that works anymore. `et gdad`

**Review by EvanBacon — APPROVED** [review]

File: packages/expo-system-ui/ios/ExpoSystemUI/ExpoSystemUIModule.swift
Comment (tsapeta) [inline packages/expo-system-ui/ios/ExpoSystemUI/ExpoSystemUIModule.swift:12]: Let's use a more expo-specific key here so it doesn't collide with something else.
```suggestion
      let color = UserDefaults.standard.integer(forKey: "ExpoSystemUI.backgroundColor")
```

File: packages/expo-system-ui/src/SystemUI.ts
Comment (alanjhughes) [reply packages/expo-system-ui/src/SystemUI.ts:42]: Yes, that'll work. I'll change that 👍 Do you think we need to add a function to clear the stored value? Reset it back to nil so it falls back to following the system theme.

Comment (tsapeta) [reply packages/expo-system-ui/src/SystemUI.ts:42]: What do you think if we just let you do that with `setBackgroundColorAsync(null)`? The getter can already return `null` if it is unset, right? It would be consistent then 🤔

Comment (alanjhughes) [reply packages/expo-system-ui/src/SystemUI.ts:42]: Yes, the getter can return null so probably makes more sense to allow the setter to accept null. I'll document the behaviour when you do that.

**Comment by brentvatne** [comment]
waiting on review from @tsapeta

File: packages/expo-system-ui/src/SystemUI.ts
Comment (tsapeta) [inline packages/expo-system-ui/src/SystemUI.ts:16]: Don't use `===` for null checks, `==` is better as it handles `undefined` too.
```suggestion
  if (color == null) {
```

**Review by tsapeta — APPROVED** [review]

**Pull request merged**
